set-client/src/main/java/eu/europeana/set/client/config/ClientConfiguration.java

```diff
@@ -1,126 +1,76 @@
 
-import eu.europeana.set.client.exception.TechnicalRuntimeException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * configuration for accessing remote api
- * 
  * @author GordeaS
- *
  */
 
 public final class ClientConfiguration {
 
+    private static final Logger LOGGER = LogManager.getLogger(ClientConfiguration.class);
+
     protected static final String SET_CLIENT_PROPERTIES_FILE = "/set-client.user.properties";
-    protected static final String PROP_SET_API_KEY = "set.api.key";
-    protected static final String PROP_SET_SERVICE_URI = "set.service.uri";
-    protected static final String PROP_OAUTH_SERVICE_URI = "oauth.service.uri";
-    protected static final String PROP_OAUTH_REQUEST_PARAMS = "oauth.token.request.params";
+    public static final String PROP_SET_API_KEY = "set.api.key";
+    public static final String PROP_SET_SERVICE_URI = "set.service.uri";
+    public static final String PROP_OAUTH_SERVICE_URI = "oauth.service.uri";
+    public static final String PROP_OAUTH_REQUEST_PARAMS = "oauth.token.request.params";
 
     private Properties properties;
-    private static ClientConfiguration singleton;
-
-    /**
-     * Hide the default constructor
-     */
-    private ClientConfiguration() {
-    }
 
     /**
-     * Accessor method for the singleton
-     * 
-     * @return
+     * Creates ClientConfiguration instance with set client properties
      */
-    public static synchronized ClientConfiguration getInstance() {
-	if (singleton == null) {
-	    singleton = new ClientConfiguration();
-	    singleton.loadProperties();
-	}
-
-	return singleton;
+    public ClientConfiguration() {
+        loadProperties(SET_CLIENT_PROPERTIES_FILE);
     }
 
     /**
-     * Laizy loading of configuration properties
+     * CConstructor to inject properties
+     * @param properties
      */
-    public synchronized void loadProperties() {
-
-	try {
-	    properties = new Properties();
-	    InputStream resourceAsStream = getClass().getResourceAsStream(SET_CLIENT_PROPERTIES_FILE);
-	    if (resourceAsStream == null) {
-		throw new TechnicalRuntimeException(
-			"No properties file found in classpath! " + SET_CLIENT_PROPERTIES_FILE);
-	    }
-	    getProperties().load(resourceAsStream);
-
-	} catch (RuntimeException | IOException e) {
-	    throw new TechnicalRuntimeException("Cannot read configuration file: " + SET_CLIENT_PROPERTIES_FILE, e);
-	}
-
+    public ClientConfiguration(Properties properties) {
+        this.properties = properties;
     }
 
-    /**
-     * provides access to the configuration properties. It is not recommended to use
-     * the properties directly, but the
-     * 
-     * @return
-     */
-    Properties getProperties() {
-	return properties;
+    private Properties loadProperties(String propertiesFile) {
+        try {
+            properties = new Properties();
+            properties.load(getClass().getResourceAsStream(propertiesFile));
```

Comment:
we should support externalization of configurations, meaning to be able to handle absolute file path using the file protocol see: https://stackoverflow.com/questions/8406025/local-file-protocol-for-java-net-url. Third parties should be able to provide  the location of configuration files. (absolute file path should use the fille:// protocol)